### DISCUSSION TITLE
Is it better to go through the list of assets once?

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#731] [Questions] [answered]
I noticed in our add_constraints! code we run through the list of assets (or a subset) to see if each constraint should be applied.
Would it be faster/more efficient to go through the list once, and for each asset apply all of the relevant constraints?

(Maybe a naïve question.)
@abelsiqueira

**Comment by abelsiqueira** [comment, marked as answer]
It's possible that it is faster. I think the SINTEF model does something like that, but they use a different structure that makes sense to work like that. Maybe when we investigate their packages we can try this out.
With out current structures it could still work, but I would investigate the other major performance issues before this.
